tests/integration/commands/dev/dev.exec.test.mjs

```diff
@@ -0,0 +1,21 @@
+import process from 'process'
+
+import { test } from 'vitest'
+
+import callCli from '../../utils/call-cli.cjs'
+import { withSiteBuilder } from '../../utils/site-builder.cjs'
+
+test('should pass .env variables to exec command', async (t) => {
+  await withSiteBuilder('site-env-file', async (builder) => {
+    builder.withEnvFile({ env: { VITEST_TEST: 'ENV_VAR' } })
```

Comment:
I think you don't have to change the env var, this should be independent of vitest (could be even `MY_SUPER_SECRET`).

I would even try to use a variable that is not used by vitest to avoid having to leak some variable that is getting set by the test runner. I think `VITEST_TEST` is set by the test runner.


```suggestion
    builder.withEnvFile({ env: { TEST: 'ENV_VAR' } })
```